gamestonk_terminal/portfolio_optimization/port_optimization_controller.py

```diff
@@ -0,0 +1,277 @@
+        print(weights)
+        print("")
+
+    def call_mkt_cap(self, other_args: List[str]):
+        weights = po_api.property_weighting(self.tickers, "marketCap", other_args)
+        print("Market Cap Weighting Weights:")
+        print(weights)
+        print("")
+
+    def call_div_yield(self, other_args: List[str]):
+        weights = po_api.property_weighting(self.tickers, "dividendYield", other_args)
+        print("Dividend Weighed Weights:")
+        print(weights)
+        print("")
+
+    def call_max_sharpe(self, other_args: List[str]):
+        weights = po_api.ef_portfolio(self.tickers, "max_sharpe", other_args)
+        print("Maximum Sharpe Weights:")
+        print(weights)
+        print("")
+
+    def call_min_vol(self, other_args: List[str]):
+        weights = po_api.ef_portfolio(self.tickers, "min_volatility", other_args)
+        print("Minimum volatility Weights:")
+        print(weights)
+        print("")
+
+    def call_eff_risk(self, other_args: List[str]):
+        weights = po_api.ef_portfolio(self.tickers, "eff_risk", other_args)
+        print("Weights for max returns at risk level")
+        print(weights)
+        print("")
+
+    def call_eff_ret(self, other_args: List[str]):
+        weights = po_api.ef_portfolio(self.tickers, "eff_ret", other_args)
+        print("Weights for min risk at target returns")
+        print(weights)
+        print("")
+
+    def call_show_ef(self, other_args):
+        po_api.show_ef(self.tickers, other_args)
+        print("")
+
+    def call_yolo(self, _):
+        # Easter egg :)
+        print("DFV YOLO")
+        print({"GME": 200})
+        print("")
+
+    @staticmethod
+    def add_stocks(self, other_args: List[str]):
+
+        """ Add ticker to current list for optimization"""
+        parser = argparse.ArgumentParser(
+            add_help=False,
+            prog="add",
+            description="""Add tickers for optimizing.""",
+        )
+        parser.add_argument(
+            "-t",
+            "--tickers",
+            dest="add_tickers",
+            type=lambda s: [str(item).upper() for item in s.split(",")],
+            default=[],
+            help="add tickers to optimzation.",
+        )
+        try:
+
+            if other_args:
+                if "-" not in other_args[0]:
+                    other_args.insert(0, "-t")
+
+            ns_parser = parse_known_args_and_warn(parser, other_args)
+            if not ns_parser:
+                return
+
+            for ticker in ns_parser.add_tickers:
+                self.tickers.add(ticker)
+
+            print(
+                f"\nCurrent Tickers: {('None', ', '.join(self.tickers))[bool(self.tickers)]}"
+            )
+            print("")
+        except Exception as e:
+            print(e)
+
+        print("")
+
+    @classmethod
+    def from_ca_menu(cls, ticker: str, similar: List[str]):
+        return cls(set([ticker] + similar))
+
+
+def menu_from_ca(ticker: str, similar: List[str]):
```

Comment:
I don't think you need to have a new controller menu here, you can just use the normal one. This makes the code a bit confusing imo, unless I'm missing something.

Reply:
If you think I should remove the jump to ca, then I can remove it.  I just did it this way so I could keep track of the loaded ticker to go back to ca.  I can also get rid of the class method in this case.

Reply:
That would be perfect, I think!